doc/nrf/ug_multiprotocol_support.rst

```diff
@@ -0,0 +1,52 @@
+.. _ug_multiprotocol_support:
+
+Multiprotocol support for Thread and Zigbee
+###########################################
+
+The :ref:`ug_thread` and :ref:`ug_zigbee` protocols support running another protocol in parallel.
+
+This is achieved with the :ref:`nrfxlib:softdevice_controller`, an implementation of the :ref:`ug_ble_controller`.
+The SoftDevice Controller in turn relies on the functionalities of ref:`Multi-Protocol Service Layer (MPSL) <nrfxlib:mpsl>` library, which provides services for multiprotocol applications and allows the 802.15.4 radio driver to negotiate for transmission timeslots with the controller.
+For more information about how the SoftDevice Controller and MPSL work together, read the :ref:`SoftDevice Controller's documentation <softdevice_controller_readme>`.
+
+The multiprotocol solution available for Thread and Zigbee is dynamic.
+This means that it allows for running several radio protocols simultaneously without the time-expensive uninitialization and initialization in-between the switching.
+Switching between protocols requires only a reinitialization of the radio peripheral, since protocols may operate on different frequencies, modulations, etc.
+
+Enabling multiprotocol support
+******************************
+
+To enable the multiprotocol support for either Thread or Zigbee, set the following Kconfig option:
+
+* :option:`CONFIG_BT_LL_SOFTDEVICE_DEFAULT`
+
+This option activates the :ref:`nrfxlib:softdevice_controller`, which by default supports the multiprotocol features.
+Disabling the SoftDevice Controller also disables the multiprotocol support.
+
+After enabling this option, the application can start configuring |BLE|'s API options alongside Thread's or Zigbee's in the respective :file:`.conf` file.
+
+Multiprotocol limitations in application development
+****************************************************
+
+Given the nature of the dynamic multiprotocol solution, take into account the following considerations when developing your Thread or Zigbee application:
+
+- Any kind of application or Thread/Zigbee stack logic can be interrupted by the |BLE| activity.
+  This is the primary difference between a Thread/Zigbee-only application and a multiprotocol one.
+  As a result, it is acceptable that the node can sometimes miss a frame and not respond with a MAC-level acknowledgment.
+- |BLE| scanning requires a lot of time to process |BLE| traffic and thus blocks 802.15.4 traffic.
+  When the scanning is used, it is recommended to use the Sleepy End Device role for Thread or Zigbee applications.
+- Be aware that flash operations require more time to complete.
+  A flash operation may be performed only in a free timeslot, so it has to wait until the current timeslot finishes (it may be a 802.15.4 timeslot or a |BLE| one).
+  Moreover, the Thread or Zigbee stacks cannot operate while the flash operation is being performed.
```

Comment:
Just as a note, at the moment, SoftDevice Contoller provides implementation of the flash driver which synchronize with it. This feature is activated if SoftDevice Contoller is enabled. The work to provide common flash radio scheduler interface is in progress.